ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java

```diff
@@ -101,6 +103,12 @@
         prepareDirectory(swapReserveDir);
         inventoryExistingReserveFiles();
         poolState = PoolState.INVENTORIED;
+      } else {
+        // Clean up reserve directory if previously created.
+        if (reserveDir != null) {
```

Comment:
Can we check `if(reserveDir != null && reserveDir.exists())` ?

Reply:
`deleteFileOrDirectory ` checks `exists ` first. So we don't need an extra check here.